apps/sms/test/marionette/mocks/mock_navigator_moz_mobile_message.js

```diff
@@ -27,7 +27,9 @@ Services.obs.addObserver(function(document) {
 
     if (handlers) {
       handlers.forEach(function(handler) {
-        handler.call(null, parameters);
+        window.setTimeout(function() {
```

Comment:
I found that the responsible code in the API is https://dxr.mozilla.org/mozilla-central/source/dom/mobilemessage/gonk/MobileMessageDB.jsm#3940-3941

So I don't think there is a magic class that does that for you :) setTimeout is good enough.